### PR TITLE
Update notes on design decisions

### PR DESCRIPTION
**Pull request by claremacrae**
## Edited

See the individual commits.

I've tried to match up notes in the '03.02 Design Decisions' folder with GitHub issues. The alignment is far from complete, but I hope it's an improvement.

### CODE REVIEW

File: 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md [hunk 1]
```diff
@@ -22,6 +24,8 @@ Suggested standard:
 
 ## Agree a standard begin/end prefix for marking 'do not edit' sections
 
+**Note:** *See [Agree a standard begin/end prefix for marking 'do not edit' sections · Issue #351](https://github.com/obsidian-community/obsidian-hub/issues/351) for discussion and implementation progress on this.*
+
 We use these markers in Author pages, and they do not seem to be well-understood:
```
Comment (lguenth) [inline 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md:29]: I stumbled over this part earlier. The "do not seem to be well-understood" seems to out of place without an explanation of *why* they are misunderstood.

Could we maybe prepend a short explanation/description here or at the top of this note on why we use these markers in general? Something like:
> We use these markers to highlight sections which should not be edited manually but are instead created automatically by our [[GitHub Actions for the Hub]].

File: 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ tags:
 
 Suggested standard:
 
+**Note:** *See the other files in this folder for progress on implementing this in each of the types of note.*
```
Comment (lguenth) [inline 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md:14]: I like these. And the status one as well. It makes this work a lot more transparent.
Just a small thing: I think it should be "types of note**s**"?

**Review by lguenth — COMMENTED** [review]
Alright, I finally managed to read this and understand the changes. This both reflects previous and on-going changes quite well. :) I'll wait for your reply on the brief suggestions I made before approving.

File: 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md (hunk 1 above)
Comment (claremacrae) [reply 00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/Content Comments.md]: That's a great point. I've pushed a change based on your text, and credited you as co-author. Thanks!

**Comment by claremacrae** [comment]
> Alright, I finally managed to read through this and understand what's being documented. This both reflects previous and on-going changes quite well. :) I'll wait for your reply on the brief suggestions I made before approving.

Very helpful suggestions - both done. Thanks @lguenth

**Review by lguenth — APPROVED** [review]
Perfect, thank you! :orange_heart: Now the intention behind these notes is more obvious for newcomers (which I still consider myself, as well!).

**Pull request merged**